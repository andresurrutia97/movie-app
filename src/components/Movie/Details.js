import React from 'react';
import PropTypes from 'prop-types';

const InfoDetails = ({ title, name }) => {
  return (
    <div className="flex flex-col items-center mx-2 md:flex-row md:mx-0 md:mr-2">
      <span className="text-sm text-white">{title}</span>
      <span className="hidden md:block md:mr-2">:</span>
      <span className="font-semibold text-white md:mr-2">{name}</span>
    </div>
  );
};

InfoDetails.propTypes = {
  title: PropTypes.string.isRequired,
  name: PropTypes.string,
};

InfoDetails.defaultProps = {
  name: '',
};
export default InfoDetails;
