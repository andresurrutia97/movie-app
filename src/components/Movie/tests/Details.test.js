import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Details from '../Details';

describe('<Details/>', () => {
  const component = <Details title="Genre" name="Horror" />;

  it('should render correctly', () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should contain a <span/> for each prop', () => {
    const wrapper = shallow(component);
    expect(wrapper.find('span').at(0).props().children).toBe('Genre');
    expect(wrapper.find('span').at(2).props().children).toBe('Horror');
  });
});
