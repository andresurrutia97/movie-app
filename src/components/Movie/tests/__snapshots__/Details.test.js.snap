// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Details/> should render correctly 1`] = `
<div
  className="flex flex-col items-center mx-2 md:flex-row md:mx-0 md:mr-2"
>
  <span
    className="text-sm text-white"
  >
    Genre
  </span>
  <span
    className="hidden md:block md:mr-2"
  >
    :
  </span>
  <span
    className="font-semibold text-white md:mr-2"
  >
    Horror
  </span>
</div>
`;
